bigtable/client/internal/endian.cc

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/internal/endian.h"
+
+#ifdef _MSC_VER
+#include <stdlib.h>
+#elif defined(__GNUC__) || defined(__clang__)
+#include <byteswap.h>
+#endif
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace internal {
+
+constexpr char c[sizeof(int)] = {1};
+constexpr bool IsBigEndian() {
+  return c[0] != 1;  // ignore different type comparison
+}
+
+/**
+ * Convert BigEndian numeric value into a string of bytes and return it.
+ */
+template <>
+std::string Encoder<bigtable::bigendian64_t>::Encode(
+    bigtable::bigendian64_t const& value) {
+  static_assert(std::numeric_limits<unsigned char>::digits ==
+                    sizeof(bigtable::bigendian64_t),
+                "This code assumes char is an 8-bit number");
+
+  char big_endian_buffer[9] = "";
```

Comment:
Did you try `std::string result(sizeof(bigendian64_t))`?

Reply:
This is not working, compilation error "no matching function for call to"